doc/source/structures/volumes_and_files/path.rst

```diff
@@ -0,0 +1,134 @@
+        :widths: 2 1 4
+
+        * - Suffix
+          - Type
+          - Description
+
+        * - :attr:`VOLUME`
+          - :struct:`Volume`
+          - Volume this path belongs to
+        * - :attr:`SEGMENTS`
+          - :struct:`List` of :struct:`String`
+          - List of this path's segments
+        * - :attr:`LENGTH`
+          - :struct:`Scalar`
+          - Number of segments in this path
+        * - :attr:`NAME`
+          - :struct:`String`
+          - Name of file or directory this path points to
+        * - :attr:`HASEXTENSION`
+          - :struct:`Boolean`
+          - True if path contains an extension
+        * - :attr:`EXTENSION`
+          - :struct:`String`
+          - This path's extension
+        * - :attr:`ROOT`
+          - :struct:`Path`
+          - Root path of this path's volume
+        * - :attr:`PARENT`
+          - :struct:`Path`
+          - Parent path
+        * - :meth:`CHANGEEXTENSION(extension)`
+          - :struct:`Path`
+          - Returns a new path with extension changed
+        * - :meth:`ISPARENT(path)`
+          - :struct:`Boolean`
+          - True if `path` is the parent of this path
+        * - :meth:`COMBINE(name1, [name2, ...])`
+          - :struct:`Path`
+          - Returns a new path created from this one
+
+.. attribute:: Path:VOLUME
+
+    :type: :struct:`Volume`
+    :access: Get only
+
+    Volume this path belongs to.
+
+.. attribute:: Path:SEGMENTS
+
+    :type: :struct:`List` of :struct:`String`
+    :access: Get only
+
+    List of segments this path contains. Segments are parts of the path separated by `/`. For example path `0:/directory/subdirectory/script.ks` contains the following segments:
+    `directory`, `subdirectory` and `script.ks`.
+
+.. attribute:: Path:LENGTH
+
+    :type: :struct:`Scalar`
+    :access: Get only
+
+    Number of this path's segments.
+
+.. attribute:: Path:NAME
+
+    :type: :struct:`String`
+    :access: Get only
+
+    Name of file or directory this path points to (same as the last segment).
+
+
+.. attribute:: Path:HASEXTENSION
+
+    :type: :struct:`Boolean`
+    :access: Get only
+
+    True if the last segment of this path has an extension.
+
+.. attribute:: Path:EXTENSION
+
+    :type: :struct:`String`
+    :access: Get only
+
+    Extension of the last segment of this path.
+
+.. attribute:: Path:ROOT
+
+    :type: :struct:`Path`
+    :access: Get only
+
+    Returns a new path that points to the root directory of this path's volume.
+
+.. attribute:: Path:PARENT
+
+    :type: :struct:`Path`
+    :access: Get only
+
+    Returns a new path that points to this path's parent. This method will throw an exception if this path does not have a parent (its length is 0).
+
+.. method:: Path:CHANGEEXTENSION(extension)
+
+    :parameter extension: :struct:`String` new path extension
+    :return: :struct:`Path`
+
+    Will return a new path with the extension of the last segment of this path replaced (or added if there's none).
+
```

Comment:
I'd be fine with `CHANGENAME` iff ``path:changename(newname)`` does exactly the same thing as ``path:parent:combine(newname)`` currently does, but not if it only works on the base name and not the extension.  (The thing @hvacengi was guessing I might have meant up above is definitely not what I wanted.)  We need to be fundamentally consistent in our terminology.  When we say "file name" elsewhere, we do include the extension (i.e. ``path:name`` returns "prog.ks" rather than just "prog", implying that when we say "name" we mean we're including the extension, not merely the base name.  Therefore ``:changename`` should be the same and the "name" it is changing should be the same "name" that ``:name`` is returning.